File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/main/MainActivityConcreteBlock.java
package com.jaynewstrom.concretesample.main;

import com.jaynewstrom.concrete.ConcreteBlock;
import com.jaynewstrom.concretesample.application.ApplicationComponent;

final class MainActivityConcreteBlock implements ConcreteBlock<MainActivityComponent> {

    private final ApplicationComponent applicationComponent;

    MainActivityConcreteBlock(ApplicationComponent applicationComponent) {
        this.applicationComponent = applicationComponent;
    }

    @Override public String name() {
        return getClass().getName();
    }

    @Override public MainActivityComponent createComponent() {
        return DaggerMainActivityComponent.builder().applicationComponent(applicationComponent).build();
    }
}

File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/main/MainActivityConcreteBlock.kt
package com.jaynewstrom.concretesample.main

import com.jaynewstrom.concrete.ConcreteBlock
import com.jaynewstrom.concretesample.application.ApplicationComponent

class MainActivityConcreteBlock(
    private val applicationComponent: ApplicationComponent
) : ConcreteBlock<MainActivityComponent> {
    override fun name(): String = javaClass.name

    override fun createComponent(): MainActivityComponent {
        return DaggerMainActivityComponent.builder().applicationComponent(applicationComponent).build()
    }
}
